### ISSUE TITLE
Explain common terms like "copyright", "patent" and "license"

### ISSUE THREAD

**Issue by strugee** [github/choosealicense.com#238]
Most people have a basic grasp of licensing-related stuff, but are hazy on the exact details of what the different terms mean. It would be nice if we could have a FAQ or something to explain all of it. I'd send a patch, but I'd like to solicit some feedback on what people think the exact format should be.

Here's a list of things we might want to put on such a page:
- "Copyright"
- "License"
- "Public Domain" (specifically the fact that waiving copyright law includes waiving your right to relicense, IIUC)
- "Patent" (especially since the distinction between a patent and a copyrighted work is lost on a lot of people)
- "Copyleft"
- Warranty disclaimers/liability (maybe?)

**Comment by bhuga** [comment]
This is a good idea, but I'd like to hear your thoughts on two concerns I have:
- These words mean different things in different jurisdictions. Can we make a definition of "public domain" or "patent" that is meaningful for everyone without misrepresenting for at least some readers?
- How do we expose this information? I think the site would become a little muddy to read if every instance of the word "license" was a link to a glossary entry, but I also feel like a link on the front page wouldn't be very helpful. By the time you knew you needed a glossary, it'd be too late.

**Comment by strugee** [comment]
> These words mean different things in different jurisdictions. Can we make a definition of "public domain" or "patent" that is meaningful for everyone without misrepresenting for at least some readers?

I think so? I'm not enough of an expert on non-US intellectual property law to say definitively, but I think if we make it general enough we can find a balance. E.g. "patent" could be defined as something along the lines of, "Something that places legal protections on ideas such that the law treats the patented idea similar to how it treats property. Patents " That blurb, while getting the main idea across, doesn't really say anything specific about the law.

Alternately, we could add a note stating that the definitions provided are based on U.S. IP law.

**Comment by JeromSar** [comment]
I think it's also important to clarify several other terms which use the word "patent".
For example, the Apache license includes a so called "patent grant", but the term is never further explained. For me, this was confusing and raised several questions:
- What is a patent?
- Does my code automatically have a patent when I publish it?
- What happens when my license includes a patent grant?
- What is restricted if the license doesn't contain a patent grant?
- What is the difference between patents and copyright?

I think an explanation for some of these could be included in the popovers that appear at the right side of the screen when hovering over the different characteristics.

**Comment by ionas** [comment]
distribution, installation, service should also be cleared.

**Comment by waldyrious** [comment]
For future reference, individual entries in this glossary should be linkable, and linked from pages on the site where they are used.

**Comment by benbalter** [comment]
Just thinking through implementation, we could have a `terms.yml` file in the `_data` folder, expose it as JSON, and have a client-side script to make any use of the word on the site a tooltip using the existing tool tip library that's already loaded.

**Comment by waldyrious** [comment]
Rather than a custom script, why not use the build system to replace all instances of those terms with HTML `<abbr>` tags?

**Comment by ionas** [comment]
@waldyrious
`<arconym>`, deprecated and not fitting.
`<abbr>`, abbreviation not deprecated but not really fitting.
What abotu `<dfn>` https://developer.mozilla.org/en/docs/Web/HTML/Element/dfn

**Comment by tuespetre** [comment]
I agree with @JeromSar; I am pretty ignorant when it comes to licensing and I don't have a clue what 'patent rights' means in this context or how they apply. I was hoping to find an explanation here. 😉

**Comment by antdog1982** [comment]
> Most people have a basic grasp of licensing-related stuff, but are hazy on the exact details of what the different terms mean. It would be nice if we could have a FAQ or something to explain all of it. I'd send a patch, but I'd like to solicit some feedback on what people think the exact format should be.
> 
> Here's a list of things we might want to put on such a page:
> 
> * "Copyright"
> * "License"
> * "Public Domain" (specifically the fact that waiving copyright law includes waiving your right to relicense, IIUC)
> * "Patent" (especially since the distinction between a patent and a copyrighted work is lost on a lot of people)
> * "Copyleft"
> * Warranty disclaimers/liability (maybe?)

Owner disc.  or creater rights/disclaimer

**Comment by antdog1982** [comment]
Copyright is your right to something you created or inherited or purchased.  These days they have so many loop holes which I dont think is the way it should be done.   If you have created or establish some sort of asset that has value of any sort that could have a great impact (change your life,have royalty's) should not be abled by any circumstance  change  ownership unless you have a  verbal  hand written contract.  That way there is no way to have conflict.  I try to read the fine lines and things can be worded in ways that can be confusion about what is going to happen if you click this or load this.  On line it should have some sort of noticeable reaction to state are you sure you will lose all rights to your ownership of if you proceed.  A copyright is yours and should be a license should not have any priority of your copyright.

**Comment by karam72** [comment]
It could be great if you can include Warranty disclaimers/liability term with one simple sentence.
